service/storage_proxy.cc

```diff
@@ -1177,10 +1177,44 @@ future<bool> paxos_response_handler::accept_proposal(const paxos::proposal& prop
 future<> paxos_response_handler::learn_decision(paxos::proposal decision, bool allow_hints) {
     tracing::trace(tr_state, "learn_decision: committing {} with cl={}", decision, _cl_for_learn);
     paxos::paxos_state::logger.trace("CAS[{}] learn_decision: committing {} with cl={}", _id, decision, _cl_for_learn);
-    std::array<std::tuple<paxos::proposal, schema_ptr, dht::token>, 1> m{std::make_tuple(std::move(decision), _schema, _key.token())};
     // FIXME: allow_hints is ignored. Consider if we should follow it and remove if not.
     // Right now we do not store hints for when committing decisions.
-    return _proxy->mutate_internal(std::move(m), _cl_for_learn, false, tr_state, _permit, _timeout);
+    future<> f_cdc = make_ready_future<>();
+    if (_schema->cdc_options().enabled()) {
+        auto update_mut = decision.update.unfreeze(_schema);
+        const auto base_tbl_id = update_mut.column_family_id();
+        std::vector<mutation> update_mut_vec;
+        update_mut_vec.push_back(std::move(update_mut));
+
+        if (_proxy->get_cdc_service() && _proxy->get_cdc_service()->needs_cdc_augmentation(update_mut_vec)) {
```

Comment:
You can move the _proxy->get_cdc_service() check earlier (with enabled()), there is no point in unfreezing the mutation if there is none. By the way, is it even possible that cdc_options().enabled() but !get_cdc_service()?

Reply:
It's not possible so let's drop check for get_cdc_service().

Reply:
done